driver-core/src/main/com/mongodb/client/gridfs/model/GridFSUploadOptions.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.gridfs.model;
+
+import org.bson.conversions.Bson;
+
+/**
+ * GridFS upload options
+ *
+ * Customizable options used when uploading files into GridFS
+ *
+ * @since 3.1
+ */
+public final class GridFSUploadOptions {
+    private Integer chunkSizeBytes;
+    private Bson metadata;
+
```

Comment:
It's simpler to omit certainly.  I hate to add more deprecated methods to a new API.